plugin/evm/limitorders/memory_database.go

```diff
@@ -136,7 +207,12 @@ func (db *InMemoryDatabase) UpdateFilledBaseAssetQuantity(quantity *big.Int, ord
 	}
 
 	if limitOrder.BaseAssetQuantity.Cmp(limitOrder.FilledBaseAssetQuantity) == 0 {
-		deleteOrder(db, orderId)
+		limitOrder.LifecycleList = append(limitOrder.LifecycleList, Lifecycle{blockNumber, FulFilled})
+	}
+
+	if quantity.Cmp(big.NewInt(0)) == -1 && limitOrder.getOrderStatus().Status == FulFilled {
```

Comment:
Need to update the orders status back to placed

Reply:
the `Status` member from the order struct has been removed and I am maintaining an order lifecycle list so we can revert to last status when a reorg happens. So after this operation, `limitOrder.LifecycleList[:len(limitOrder.LifecycleList)-1]` will contain `Placed`